finite-state/shared/src/main/scala/fs2/data/mft/MFTBuilder.scala

```diff
@@ -0,0 +1,58 @@
+package fs2.data.mft
+
+import scala.collection.mutable.ListBuffer
+
+class MFTBuilder[InTag, OutTag] private[mft] {
+  self =>
+
+  private var initial = 0
+
+  private val states = new ListBuffer[StateBuilder]
+
+  final class StateBuilder private[mft] (val q: Int, val nargs: Int) {
+    var rules = new ListBuffer[(EventSelector[InTag], Rhs[OutTag])]
```

Comment:
Why do we need a `var` of a mutable collection here?

Reply:
Indeed this is leftover from previous attempt.